plugins/view-transitions/includes/class-plvt-view-transition-animation.php

```diff
@@ -0,0 +1,283 @@
+	 *     @type callable|null $get_stylesheet_callback     Callback to get the stylesheet for the animation, as
+	 *                                                      inline CSS. This can be used if the animation CSS requires
+	 *                                                      further preparation other than simply loading its
+	 *                                                      stylesheet from the animation's corresponding CSS file.
+	 *                                                      Default null.
+	 * }
+	 * @param array<string, mixed> $default_args Optional. Default animation arguments. Default empty array.
+	 *
+	 * @throws InvalidArgumentException Thrown if the slug or an alias is invalid.
+	 */
+	public function __construct( string $slug, array $config, array $default_args = array() ) {
+		if ( ! $this->is_valid_slug( $slug ) ) {
+			throw new InvalidArgumentException(
+				sprintf(
+					/* translators: %s: invalid slug */
+					esc_html__( 'The animation slug "%s" is invalid.', 'view-transitions' ),
+					esc_html( $slug )
+				)
+			);
+		}
+
+		$this->slug = $slug;
+
+		$this->apply_config( $config );
+
+		$this->default_args = $default_args;
+	}
+
+	/**
+	 * Gets the unique animation slug.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return string Unique animation slug.
+	 */
+	public function get_slug(): string {
+		return $this->slug;
+	}
+
+	/**
+	 * Gets the unique aliases for the animation, if any.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return string[] Unique aliases for the animation, or empty array if none.
+	 */
+	public function get_aliases(): array {
+		return $this->aliases;
+	}
+
+	/**
+	 * Gets the animation stylesheet, as inline CSS.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param string               $alias Optional. Slug or alias to reference the animation with. May be used to alter
+	 *                                    the animation's behavior. Default is the animation's slug.
+	 * @param array<string, mixed> $args  Optional. Animation arguments. Default is the animation's default arguments.
+	 * @return string Animation stylesheet, as inline CSS, or empty string if none.
+	 */
+	public function get_stylesheet( string $alias = '', array $args = array() ): string {
+		if ( $this->use_stylesheet ) {
+			// phpcs:ignore WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+			$css = file_get_contents( plvt_get_asset_path( "css/view-transition-animation-{$this->slug}.css" ) );
+			if ( false === $css || '' === $css ) {
+				// This clause should never be entered, but is needed to please PHPStan. Can't hurt to be safe.
+				return '';
+			}
+		}
+		if ( is_callable( $this->get_stylesheet_callback ) ) {
+			if ( '' === $alias ) {
+				$alias = $this->slug;
+			}
+			$args = wp_parse_args( $args, $this->default_args );
+			return (string) call_user_func_array(
+				$this->get_stylesheet_callback,
+				isset( $css ) ? array( $css, $alias, $args ) : array( $alias, $args )
```

Comment:
```suggestion
				isset( $css ) ? array( $css, $alias, $args ) : array( '', $alias, $args )
```

I was thinking in the current approach, the callback function receives different parameter counts, which wouldn't be a problem if the parameter being omitted was at the end. However, removing the first parameter changes the sequence of all parameters passed, which feels wrong. I think just sending an empty string there would make it a consistent API.

Reply:
Fair point, I'll update this.

Reply:
Fixed in https://github.com/WordPress/performance/pull/2016/commits/673d14bce8b6b15263e9ec3f805db74e82cd80ab